docs/journal/02-exploring-recent-events.md

```diff
@@ -1,7 +1,122 @@
+## Common requirements
+
+- Queries are ad hoc. You may not know in advance what queries are needed.
+- Queries must execute reasonably quickly, so the user can explore the data. This differs from running a batch query at some fixed interval.
+- Queries typically involve a time range: “What was the average temperature during this 20-minute window?” 
+- Queries might need relational data: “What was the average temperature across all sensors in building 25?” 
+- Relational data might be stored separately from event data; for example, a device registry that maps device IDs to buildings.
+- Ad hoc exploration *tends* to target more recent data, rather than the entire set of historical data. The exact definition of “recent” depends on your particular scenario. It might mean the past 24 hours, the past week, or the past month.
+- The system must aggregate results into a meaningful view. IoT solutions can generate a *lot* of data, even across a relatively small time slice, so you need a way for a human to understand the results.
+- Ad hoc exploration is generally "user initiated," meaning the user submits a query (or perhaps logs into a dashboard), versus having the system detect a condition and notify the user.
+
+> We don't include critical alerts under ad hoc exploration. Ad hoc queries should execute quickly, but not necessarily in real time. (See the [real-time processing][real-time-milestone] milestone.) 
+
+
+## Fabrikam requirements
+
+:memo: For more about our Fabrikam scenario, see [Introducing the Journey](00-introducing-the-journey.md). 
+
+Here are the specific requirements for our scenario:
+
+- Fabrikam needs to query temperature data by building or room number, over a specified time window. For example, "What was the average temperature in building 25 over the past hour?"
+
+**[What is the maximum time window that we are targeting? i.e., past day? past week?]**
+
+**[Other requirements?]**
+
+## Solutions
+
+We explored the following solutions:
+
+- Azure Stream Analytics + Azure SQL
+- Elastic Search
+
+> **Time series databases** are another interesting possibility. As the name implies, time series  databases are optimized for time series data, which seems like a natural fit for IoT. We have not investigated this approach yet. 
+
+> We also looked at writing the raw events directly to SQL, PowerBI, or DocumentDB. However, none of these has enough throughput and/or capacity for our requirements. (See issues [88], [89], [90].)
+
+### Stream Analytics + SQL
+
+[Azure Stream Analytics] is a real-time stream processing service. It uses an SQL-like language to process incoming data. Unlike traditional SQL, Stream Analytics runs queries over an event stream, making it easy to analyze events within a temporal window.     
+
+In this approach, we use Stream Analytics to aggregate the event data in real time and store the results in SQL. The aggregated result is small enough that SQL can handle the load.
```

Comment:
"small" meaning number of results per second, not necessarily the amount of data.